omc3/model/accelerators/lhc.py

```diff
@@ -486,12 +488,48 @@ def _get_madx_main_sequence_loading(self) -> str:
                 "has to be specified (--lhcmode option missing?)."
             )
 
-    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str]) -> str:
+    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str], update_dpp: bool = False) -> str:
         madx_script = self.get_base_madx_script()
-        for corr_file in corr_files:
+
+        # First set the dpp to the value in the accelerator model
+        madx_script += f"{ORBIT_DPP} = {self.dpp};\n"
+
+        for corr_file in corr_files:  # Load the corrections, can also update ORBIT_DPP
             madx_script += f"call, file = '{str(corr_file)}';\n"
-        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {self.dpp});\n"
+        
+        if update_dpp: # If we are doing orbit correction, we need to ensure that a correct and matching is done (in get_update_deltap_script)
+            madx_script += self.get_update_deltap_script(deltap=ORBIT_DPP)
+
+        madx_script += f'exec, do_twiss_elements(LHCB{self.beam}, "{str(outpath)}", {ORBIT_DPP});\n'
         return madx_script
+    
+    def get_update_deltap_script(self, deltap: float | str) -> str:
+        if not isinstance(deltap, str):
+            deltap = f"{deltap:.15e}"
```

Comment:
Maybe we can also check the `deltap` is some form of number and raise if not? I don't think we'd end up in this scenario but it could help diagnose since this is called quite deep into some code loops.

Not mandatory, feel free to ignore and resolve.

Reply:
```suggestion
        if isinstance(deltap, str):
            deltap = f"{deltap:.15e}"
```

Reply:
Yes but maybe `isinstance(a, (int, float))` since an `int` would not trigger that check